src/perl/perl-signals.c

```diff
@@ -291,6 +300,14 @@ static void perl_call_signal(PERL_SCRIPT_REC *script, SV *func,
 			/* "simple irssi object" - any struct that has
 			   int type; as it's first variable (dcc) */
 			perlarg = simple_iobject_bless((SERVER_REC *) arg);
+		} else if (g_strcmp0(rec->args[n], "gbytes") == 0) {
+			GBytes *tmp;
+			const char *data;
+			STRLEN data_len;
+
+			tmp = arg;
+			data = g_bytes_get_data(tmp, &data_len);
+			perlarg = newSVpvn(data, data_len);
```

Comment:
I'm not sure of the lifecycle of this perl argument. Should I be allocating from the heap and memcpy this for Perl?